### ISSUE TITLE
Handle mWeb permission denied flows

### ISSUE THREAD

**Issue by Julesssss** [Expensify/App#42978]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Holding on [this issue](https://github.com/Expensify/App/issues/37736) and [this PR](https://github.com/Expensify/App/pull/39875)**

## Action Performed:
- _Location permission should not yet be enabled_
- Create a distance request
- (A) Should be prompted for location permission with our customer modal
- Tap continue and the OS permission prompt should display
- Reject it and repeat a few times
- (B) Eventually the OS permission prompt will stop showing, instead our 'Allow location access' modal should be displayed with the Camera Pigeon icon
- Tap 'Settings'
- You should be routed to the relevant OS permission settings page

## Actual Result:
- (A) - You see a small sentence that is clickable instead of our modal (example for different flow shown below), and we do not show a permission prompt within our app
- (B) - We never show the secondary modal after the permission has been denied

<img width="794" alt="Screenshot 2024-06-03 at 09 56 37" src="https://github.com/Expensify/App/assets/10736861/89127e7a-8756-4abf-967c-d17797b924bd">

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<img width="372" alt="Screenshot 2024-06-03 at 09 44 06" src="https://github.com/Expensify/App/assets/10736861/e51f7aac-39e1-4d64-bf1f-1de5d3957493">

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by garrettmknight** [comment]
Can you link the hold issue here? Happy to help manage if it's important.

**Comment by Julesssss** [comment]
Holding on [this issue](https://github.com/Expensify/App/issues/37736) and [this PR](https://github.com/Expensify/App/pull/39875) -- I'm splitting out a small part of that flow

**Comment by Julesssss** [comment]
Thanks, I don't think there's much to do as [the issue](https://github.com/Expensify/App/issues/37736) is close to complete, but there is no BugZero team member assigned to the issue. So assigning yourself there would be helpful 👍  

We're holding on [this issue](https://github.com/Expensify/App/issues/37736) and [this PR](https://github.com/Expensify/App/pull/39875) -- in this issue I'm splitting out a small, somewhat unrelated part of that flow to help speed up the implementation

**Comment by Julesssss** [comment]
Still held

**Comment by Julesssss** [comment]
HELD

**Comment by Julesssss** [comment]
Blockers are still held

**Comment by Julesssss** [comment]
Still held

**Comment by Julesssss** [comment]
Other flow issues in progress

**Comment by Julesssss** [comment]
Not a priority yet

**Comment by Julesssss** [comment]
Still low

**Comment by Julesssss** [comment]
Low priority. We are handling the delaed prompt though, so this should probably follow that task

**Comment by Julesssss** [comment]
Low priority

**Comment by Julesssss** [comment]
We are currently adding a location permission delay. Once that is complete we can think about this improvement

**Comment by Julesssss** [comment]
Still held on above

**Comment by Julesssss** [comment]
We decided to remove our prompt from the web flow entirely.
